hadoop-hdds/common/src/main/java/com/google/protobuf/XmlToMarkdown.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.google.protobuf;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * Represents a property in the XML file.
+ */
+class Property {
+  private final String name;
+  private final String value;
+  private final String tag;
+  private final String description;
+
+  Property(String name, String value, String tag, String description) {
+    this.name = name;
+    this.value = value;
+    this.tag = tag;
+    this.description = description;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public String getValue() {
+    return value;
+  }
+
+  public String getTag() {
+    return tag;
+  }
+
+  public String getDescription() {
+    return description;
+  }
+}
+
+/**
+ * Utility class for converting XML properties from ozone-default.xml and
+ * ozone-default-generated.xml into Markdown format. The converted content
+ * is saved to Configurations.md.
+ */
+public final class XmlToMarkdown {
+
+  private XmlToMarkdown() {
+    // Prevent instantiation
+  }
+
+  public static void main(String[] args) {
+    try {
+      // List of XML files to process
+      String[] xmlFiles = {
+          "hadoop-hdds/common/src/main/resources/ozone-default.xml",
+          "hadoop-hdds/config/target/test-classes/ozone-default-generated.xml"
```

Comment:
`ozone-default-generated.xml` should be picked up from all modules.  However, we need to skip `test-classes`, those configs are only for testing purposes.

It would be nice to find these without having to hard-code the paths.

```
hadoop-hdds/client/target/classes/ozone-default-generated.xml
hadoop-hdds/common/target/classes/ozone-default-generated.xml
hadoop-hdds/container-service/target/classes/ozone-default-generated.xml
hadoop-hdds/framework/target/classes/ozone-default-generated.xml
hadoop-hdds/server-scm/target/classes/ozone-default-generated.xml
hadoop-ozone/common/target/classes/ozone-default-generated.xml
hadoop-ozone/csi/target/classes/ozone-default-generated.xml
hadoop-ozone/ozone-manager/target/classes/ozone-default-generated.xml
hadoop-ozone/recon-codegen/target/classes/ozone-default-generated.xml
hadoop-ozone/recon/target/classes/ozone-default-generated.xml
```

Since `hdds-common` and `hdds-docs` are compiled before most of those other modules, we'll need to find another way to run this converter.